test/e2e/tolerations_test.go

```diff
@@ -0,0 +1,207 @@
+
+	logrus.Infof("Creating jaeger services for tolerations test. jaeger-cr:%s, namespace:%s", jaegerInstanceName, namespace)
+	createESSelfProvDeployment(jaegerCR, jaegerInstanceName, namespace)
+	defer undeployJaegerInstance(jaegerCR)
+
+	ProductionSmokeTest(jaegerInstanceName)
+
+	// verify tolerations
+
+	collectorDeployments := getDeployments(namespace, fmt.Sprintf("app=jaeger,app.kubernetes.io/component=collector,app.kubernetes.io/instance=%s", jaegerInstanceName))
+	require.Equal(t, 1, len(collectorDeployments), "Collector deployments count not matching")
+	collectorDeployment := collectorDeployments[0]
+	require.Equal(t, collectorReplicasCount, collectorDeployment.Status.ReadyReplicas, "Collector deployment replicas count not matching")
+	suite.verifyTolerations(collectorDeployment.Name, tolerationsCollector, collectorDeployment.Spec.Template.Spec.Tolerations)
+
+	queryDeployments := getDeployments(namespace, fmt.Sprintf("app=jaeger,app.kubernetes.io/component=query,app.kubernetes.io/instance=%s", jaegerInstanceName))
+	require.Equal(t, 1, len(queryDeployments), "Query deployments count not matching")
+	queryDeployment := queryDeployments[0]
+	require.Equal(t, queryReplicasCount, queryDeployment.Status.ReadyReplicas, "Query deployment replicas count not matching")
+	suite.verifyTolerations(queryDeployment.Name, tolerationsQuery, queryDeployment.Spec.Template.Spec.Tolerations)
+
+	if includeESSelfProvision {
+		esDeployments := getDeployments(namespace, "component=elasticsearch")
+		require.Equal(t, esNodeCount, int32(len(esDeployments)), "Elasticsearch deployments count not matching")
+		for index := 0; index < len(esDeployments); index++ {
+			esDeployment := esDeployments[index]
+			require.Equal(t, int32(1), esDeployment.Status.ReadyReplicas, "Elasticsearch deployment replicas count not matching")
+			suite.verifyTolerations(esDeployment.Name, tolerationsES, esDeployment.Spec.Template.Spec.Tolerations)
+		}
+	}
+}
+
+func (suite *TolerationsTestSuite) getTolerations(prefix string) []v1.Toleration {
+	tolerations := []v1.Toleration{
+		{Key: fmt.Sprintf("%s_equal_key1", prefix), Operator: v1.TolerationOpEqual, Value: "value1", Effect: v1.TaintEffectNoExecute},
+		{Key: fmt.Sprintf("%s_equal_key2", prefix), Operator: v1.TolerationOpEqual, Value: "value2", Effect: v1.TaintEffectNoSchedule},
+		{Key: fmt.Sprintf("%s_equal_key3", prefix), Operator: v1.TolerationOpEqual, Value: "value3", Effect: v1.TaintEffectPreferNoSchedule},
+
+		{Key: fmt.Sprintf("%s_exists_key1", prefix), Operator: v1.TolerationOpExists, Effect: v1.TaintEffectNoExecute},
+		{Key: fmt.Sprintf("%s_exists_key2", prefix), Operator: v1.TolerationOpExists, Effect: v1.TaintEffectNoSchedule},
+		{Key: fmt.Sprintf("%s_exists_key3", prefix), Operator: v1.TolerationOpExists, Effect: v1.TaintEffectPreferNoSchedule},
+	}
+	return tolerations
+}
+
+func (suite *TolerationsTestSuite) verifyTolerations(deploymentName string, expectedList []v1.Toleration, actualList []v1.Toleration) {
```

Comment:
I wouldn't pass in deployment name here if it's only used for debugging.  Add a logrus.Debug statement in the caller instead.